sparkmagic/sparkmagic/tests/test_remotesparkmagics.py

```diff
@@ -269,6 +269,99 @@ def test_run_cell_command_exception():
 
 
 @with_setup(_setup, _teardown)
+def test_run_spark_command_parses():
+    run_cell_method = MagicMock()
+    run_cell_method.return_value = (True, "")
+    spark_controller.run_command = run_cell_method
+
+    command = "-s"
+    name = "sessions_name"
+    context = "-c"
+    context_name = "spark"
+    meth = "-m"
+    method_name = "sample"
+    line = " ".join([command, name, context, context_name, meth, method_name])
+    cell = "cell code"
+
+    result = magic.spark(line, cell)
+
+    run_cell_method.assert_called_once_with(Command(cell), name)
+
+
+@with_setup(_setup, _teardown)
+def test_run_spark_with_store_command_parses():
+    run_cell_method = MagicMock()
+    run_cell_method.return_value = (True, "")
+    spark_controller.run_command = run_cell_method
+
+    command = "-s"
+    name = "sessions_name"
+    context = "-c"
+    context_name = "spark"
+    meth = "-m"
+    method_name = "sample"
+    output = "-o"
+    output_var = "var_name"
+    line = " ".join([command, name, context, context_name, meth, method_name, output, output_var])
+    cell = "cell code"
+
+    result = magic.spark(line, cell)
+
+    run_cell_method.assert_any_call(Command(cell), name)
+    run_cell_method.assert_any_call(SparkStoreCommand(cell, output_var, samplemethod=method_name), name)
+
+
+@with_setup(_setup, _teardown)
+def test_run_spark_command_exception():
+    run_cell_method = MagicMock()
```

Comment:
I think that there should be unit tests that test only what this class does. Since this class is not actually executing anything against the spark_controller when running spark (sparkmagicsbase is), this should not test an exception there. Instead, it should just test that the exception is caught when it is thrown by `execute_spark`. If this is not clear, we can discuss on Monday.